keras_hub/src/layers/preprocessing/image_converter.py

```diff
@@ -294,9 +294,10 @@ def _expand_non_channel_dims(self, value, inputs):
             # device (potentially GPU) after preprocessing.
             if keras.backend.backend() == "torch" and self.image_size is None:
                 return ops.expand_dims(value, broadcast_dims).cpu()
-            return ops.expand_dims(value, broadcast_dims)
+            expanded = ops.expand_dims(value, broadcast_dims)
+            return ops.cast(expanded, inputs.dtype)
         else:
-            return np.expand_dims(value, broadcast_dims)
+            return np.expand_dims(value, broadcast_dims).astype(inputs.dtype)
```

Comment:
You might to call standardize dtype here, looks like tf dtypes are causing issues.

Reply:
It's weird that the tests are not failing locally